modules/dns/dns_api.rb

```diff
@@ -35,16 +38,35 @@ class Api < ::Sinatra::Base
     end
 
     delete "/:value" do
-      type = params[:value] =~ /\.(in-addr|ip6)\.arpa$/ ? "PTR" : "A"
+      name = params[:value]
+      type = name =~ /\.(in-addr|ip6)\.arpa$/ ? "PTR" : "A"
 
       begin
-        server.remove_a_record(params[:value]) if type == 'A'
-        server.remove_ptr_record(params[:value]) if type == 'PTR'
+        validate_dns_name!(name)
+
+        case type
+        when 'A'
+          server.remove_a_record(name)
+        when 'PTR'
+          validate_reverse_dns_name!(name)
+          server.remove_ptr_record(name)
+        else
+          log_halt(400, "unrecognized 'type' parameter: #{type}")
+        end
       rescue Proxy::Dns::NotFound => e
         log_halt 404, e
       rescue => e
         log_halt 400, e
       end
     end
+
+    def validate_dns_name!(name)
+      raise Proxy::Dns::Error.new("Invalid DNS name #{name}") unless name =~ /^([a-zA-Z0-9]([-a-zA-Z0-9]+)?\.?)+$/
```

Comment:
I think this is ok: 
 - For A/AAAA record creation this is more of a precaution, as the server is going to do its own verification
 - Not sure how valuable this is for PTR record creation -- fqdn is used for detection of duplicate records only
 - Similarly for record deletion -- I doubt that servers perform fqdn validation, as all they need is a match/no match against known host names...